src/lib/rest/OrionError.cpp

```diff
@@ -144,62 +151,33 @@ std::string OrionError::render(ConnectionInfo* ciP, const std::string& _indent)
 }
 
 
+
 /* ****************************************************************************
 *
 * OrionError::render -
 *
-* FIXME P3: OrionError() render method should be unified (probably the class itself needs
-* a good refactoring, to live in the NGSIv1-NGSIv2 dual world...)
+* This method removes any whitespace in the reasonPhrase field, i.e.
+* transforms "Not Found" to "NotFound".
 *
-* Simplified version, without ciP
+* It is used by smartRender method, in order to prepare to render in API v2 case
 *
 */
-std::string OrionError::render(const std::string& _indent, const std::string apiVersion)
+void OrionError::shrinkReasonPhrase(void)
 {
-  //
-  // For API version 2 this is pretty easy ...
-  //
-  if (apiVersion == "v2")
-  {
-    if (details == "Already Exists")
-    {
-      details = "Entity already exists";
-    }
+  char buf[80];  // 80 should be enough to hold any reason phrase
 
-    reasonPhrase = errorStringForV2(reasonPhrase);
-    return "{" + JSON_STR("error") + ":" + JSON_STR(reasonPhrase) + "," + JSON_STR("description") + ":" + JSON_STR(details) + "}";
-  }
+  strncpy(buf, reasonPhrase.c_str(), 80);
 
-
-  //
-  // A little more hairy for API version 1
-  //
-
-  std::string  out           = "";
-  std::string  tag           = "orionError";
-  std::string  initialIndent = _indent;
-  std::string  indent        = _indent;
-
-  //
-  // OrionError is NEVER part of any other payload, so the JSON start/end braces must be added here
-  //
-
-
-  out     = initialIndent + "{\n";
-  indent += "  ";
-
-  out += startTag1(indent, tag);
-  out += valueTag(indent + "  ", "code",          code,         true);
-  out += valueTag1(indent + "  ", "reasonPhrase",  reasonPhrase, details != "");
-
-  if (details != "")
+  // See: http://stackoverflow.com/questions/1726302/removing-spaces-from-a-string-in-c
+  char* i = buf;
+  char* j = buf;
+  while(*j != 0)
   {
-    out += valueTag1(indent + "  ", "details",       details);
+    *i = *j++;
+    if(*i != ' ')
+      i++;
   }
+  *i = 0;
 
-  out += endTag(indent);
-
-  out += initialIndent + "}\n";
-
-  return out;
+  reasonPhrase = std::string(buf);
 }
```

Comment:
This was the preferred option but some problems (now described in function preamble in 68edae4cd8e5c7022f33cb9a957dd56dd33d835a) avoid using it. Thus, NTC.